File: PHP/Petshop.php
<?php
session_start();

class Petshop {
    private $produk;
    private $id;
    private $nama;
    private $kategori;
    private $harga;
    private $gambar;

    public function getId() {
        return $this->id;
    }

    public function setId($id) {
        $this->id = $id;
    }

    public function getNama() {
        return $this->nama;
    }

    public function setNama($nama) {
        $this->nama = htmlspecialchars($nama);
    }

    public function getKategori() {
        return $this->kategori;
    }

    public function setKategori($kategori) {
        $this->kategori = htmlspecialchars($kategori);
    }

    public function getHarga() {
        return $this->harga;
    }

    public function setHarga($harga) {
        $this->harga = floatval($harga);
    }

    public function getGambar() {
        return $this->gambar;
    }

    public function setGambar($gambar) {
        $this->gambar = filter_var($gambar, FILTER_SANITIZE_URL);
    }

    public function __construct() {
        if (!isset($_SESSION['produk'])) {
            $_SESSION['produk'] = [];
        }
        $this->produk = &$_SESSION['produk'];
    }

    public function menambahkan($nama, $kategori, $harga, $gambar) {
        $this->id = count($this->produk) + 1;
        $this->nama = htmlspecialchars($nama);
        $this->kategori = htmlspecialchars($kategori);
        $this->harga = floatval($harga);
        $this->gambar = filter_var($gambar, FILTER_SANITIZE_URL);

        $this->produk[] = [
            "id" => $this->id,
            "nama" => $this->nama,
            "kategori" => $this->kategori,
            "harga" => $this->harga,
            "gambar" => $this->gambar
        ];
        return "Produk berhasil ditambahkan!";
    }

    public function menampilkan() {
        return $this->produk;
    }

    public function menghapus($id) {
        foreach ($this->produk as $key => $p) {
            if ($p["id"] == $id) {
                unset($this->produk[$key]);
                return "Produk berhasil dihapus!";
            }
        }
        return "Produk tidak ditemukan!";
    }

    public function mencari($nama) {
        $hasil = [];
        foreach ($this->produk as $p) {
            if (stripos($p["nama"], $nama) !== false) {
                $hasil[] = $p;
            }
        }
        return $hasil;
    }

    public function mengubah($id, $nama, $kategori, $harga, $gambar) {
        foreach ($_SESSION['produk'] as &$p) {
            if ($p["id"] == $id) {
                $p["nama"] = htmlspecialchars($nama);
                $p["kategori"] = htmlspecialchars($kategori);
                $p["harga"] = floatval($harga);
                $p["gambar"] = filter_var($gambar, FILTER_SANITIZE_URL);
                return "Produk berhasil diperbarui!";
            }
        }
        return "Produk tidak ditemukan!";
    }
}
?>

File: PYTHON/Petshop.py
class Petshop:
    def __init__(self, id=1, nama_produk="", kategori_produk="", harga_produk=0):
        self.id = id
        self.nama_produk = nama_produk
        self.kategori_produk = kategori_produk
        self.harga_produk = harga_produk

    def set_id(self, id):
        self.id = id

    def set_nama_produk(self, nama_produk):
        self.nama_produk = nama_produk

    def set_kategori_produk(self, kategori_produk):
        self.kategori_produk = kategori_produk

    def set_harga_produk(self, harga_produk):
        self.harga_produk = harga_produk

    def get_id(self):
        return self.id

    def get_nama_produk(self):
        return self.nama_produk

    def get_kategori_produk(self):
        return self.kategori_produk

    def get_harga_produk(self):
        return self.harga_produk

    def menambahkan(self, list_produk):
        print("\nMenambahkan Produk")
        nama_produk = input("Nama Produk: ")
        kategori_produk = input("Kategori Produk: ")
        harga_produk = int(input("Harga Produk: "))

        produk = Petshop(len(list_produk) + 1, nama_produk, kategori_produk, harga_produk)
        list_produk.append(produk)
        print("Produk berhasil ditambahkan.")

    def menampilkan(self, list_produk):
        print("\nMenampilkan Semua Produk")
        for produk in list_produk:
            print(f"ID: {produk.get_id()}\nNama Produk: {produk.get_nama_produk()}\nKategori Produk: {produk.get_kategori_produk()}\nHarga Produk: {produk.get_harga_produk()}\n")

    def mengubah(self, list_produk):
        id = int(input("\nID Produk yang ingin diubah: "))
        for produk in list_produk:
            if produk.get_id() == id:
                produk.set_nama_produk(input("Nama Produk: "))
                produk.set_kategori_produk(input("Kategori Produk: "))
                produk.set_harga_produk(int(input("Harga Produk: ")))
                print("Produk berhasil diubah.")
                return
        print("Produk tidak ditemukan.")

    def menghapus(self, list_produk):
        id = int(input("\nID Produk yang ingin dihapus: "))
        for produk in list_produk:
            if produk.get_id() == id:
                list_produk.remove(produk)
                print("Produk berhasil dihapus.")
                return
        print("Produk tidak ditemukan.")

    def mencari(self, list_produk):
        nama_produk = input("\nNama Produk yang ingin dicari: ")
        for produk in list_produk:
            if produk.get_nama_produk().lower() == nama_produk.lower():
                print(f"ID: {produk.get_id()}\nNama Produk: {produk.get_nama_produk()}\nKategori Produk: {produk.get_kategori_produk()}\nHarga Produk: {produk.get_harga_produk()}")
                return
        print("Produk tidak ditemukan.")

